tests/EventListener/AutoTransitionListenerTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace tests\Libero\ContentApiBundle\EventListener;
+
+use Libero\ContentApiBundle\EventListener\AutoTransitionListener;
+use LogicException;
+use PHPUnit\Framework\TestCase;
+use stdClass;
+use Symfony\Component\Workflow\Event\Event;
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\Transition;
+use Symfony\Component\Workflow\Workflow;
+
+final class AutoTransitionListenerTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function it_applies_the_first_available_transition() : void
+    {
+        $marking = new Marking();
+        $subject = new stdClass();
+        $transition1 = new Transition('transition1', 'place1', 'place2');
+        $transition2 = new Transition('transition2', 'place3', 'place4');
+        $transition3 = new Transition('transition3', 'place5', 'place6');
+        $transition4 = new Transition('transition4', 'place7', 'place8');
+
+        $workflow = $this->createMock(Workflow::class);
+        $workflow->method('getEnabledTransitions')
+            ->with($subject)
+            ->willReturn([$transition2, $transition3, $transition4]);
+        $workflow->expects($this->exactly(2))
+            ->method('can')
+            ->withConsecutive([$subject, 'transition2'], [$subject, 'transition3'])
```

Comment:
many `with()` in the same test is the equivalent of having many assertions, consider splitting the test into different ones if this class grows in responsibilities (tests would be similar in what they do, but not in what they check)

Reply:
Not ideal, would rather not mock this but complicated to set up. I suspect we'll move to a proper queue rather than running through it in one process so will leave for now.